### PR TITLE
[CODE_GEN] Added `remote_store.restore` action

### PR DESCRIPTION
**Pull request by nhtruong**
### Description
All files, excepts `.rubocop.yml` and `CHANGELOG.md`, were generated by the [API generator](https://github.com/nhtruong/opensearch-ruby-api-generator) I'm working on. Once this has been merged, I'll follow up with a PR for the generator into this repo.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,7 @@ Inspired from [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 
 ## [Unreleased]
 ### Added
+- `remote_store.restore` action ([#176](https://github.com/opensearch-project/opensearch-ruby/pull/176))
```
Comment (dblock) [inline CHANGELOG.md:6]: Say `Added ...`

File: spec/opensearch/api/actions/remote_store/restore_spec.rb [hunk 2]
```diff
@@ -0,0 +1,46 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+#
+# Modifications Copyright OpenSearch Contributors. See
```
Comment (dblock) [inline spec/opensearch/api/actions/remote_store/restore_spec.rb:7]: The modifications header is not needed with only the above. It's for code that came from APLv2 Elastic that had `Licensed to Elasticsearch B.V. under ...`.

**Review by dblock — APPROVED** [review]
Just a nit on the CHANGELOG pls.

**Review by dblock — APPROVED** [review]

**Comment by nhtruong** [comment]
@VachaShah I'm happy with what we've got so far with this generated code. Going to do some refactoring before creating the PR to move the generator here. We can merge this now. Thanks

**Review by VachaShah — APPROVED** [review]
@nhtruong This looks great! Since this is an experimental API, are there any plans to mark it as such in the client?

**Comment by nhtruong** [comment]
@VachaShah We will need to include this info in the API spec. Something like `x-experimental: true`. Then the generator will look for that flag to display the warning. Thanks for pointing that out!

**Pull request merged**